### ISSUE TITLE
"unused import" could be more helpful ("nothing to export")

### ISSUE THREAD

**Issue by arifd** [rust-lang/rust#123855]
### Code

```Rust
mod misc {
    const WOW: usize = 1;
}

pub use misc::*;
```

### Current output

```Shell
warning: unused import: `misc::*`
 --> src/main.rs:6:9
  |
6 | pub use misc::*;
  |         ^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default

warning: `playground` (bin "playground") generated 1 warning (run `cargo clippy --fix --bin "playground"` to apply 1 suggestion)
```

### Desired output

```Shell
Either an additional hint saying why this lint is complaining "There is nothing to export"

Or a new lint or something.
```

### Rationale and extra context

In more complex codebases it's not immediately obvious what the problem is. I'd prefer a message like "nothing to export".

### Rust Version

```Shell
rustc 1.75.0 (82e1608df 2023-12-21)
binary: rustc
commit-hash: 82e1608dfa6e0b5569232559e3d385fea5a93112
commit-date: 2023-12-21
host: aarch64-apple-darwin
release: 1.75.0
LLVM version: 17.0.6
```

**Comment by eggyal** [comment]
I dont think the issue is or diagnostic should be about what is exported here.

A wildcard `use` declaration that doesn't actually bind any visible paths should error on that ground alone.  This, I think, is the issue that needs addressing here.

Irrespective of any wildcard, if a `use` declaration binds some names that are neither used nor reexported, that's a different error (the currently displayed one).
